### ISSUE TITLE
ValueError when using plot function with projection

### ISSUE THREAD

**Issue by bugsuse** [pydata/xarray#5014]
I want to plot a map using `.plot` function but raise ValueError, and the code as follows, 

```python
air = xr.tutorial.open_dataset("air_temperature").air

air.isel(time=[0, 4]).plot(
    transform=ccrs.PlateCarree(),
    col="time",
    subplot_kws={"projection": ccrs.PlateCarree()},
)
```

```python
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-186-e81a00049224> in <module>
----> 1 data['air'].isel(time=[0, 4]).plot(
      2     transform=ccrs.PlateCarree(),
      3     col='time',
      4     subplot_kws={"projection": ccrs.PlateCarree()},
      5 )

~/tools/miniconda3/envs/satpy_himawari/lib/python3.8/site-packages/xarray/plot/plot.py in __call__(self, **kwargs)
    442 
    443     def __call__(self, **kwargs):
--> 444         return plot(self._da, **kwargs)
    445 
    446     # we can't use functools.wraps here since that also modifies the name / qualname

~/tools/miniconda3/envs/satpy_himawari/lib/python3.8/site-packages/xarray/plot/plot.py in plot(darray, row, col, col_wrap, ax, hue, rtol, subplot_kws, **kwargs)
    195     kwargs["ax"] = ax
    196 
--> 197     return plotfunc(darray, **kwargs)
    198 
    199 

~/tools/miniconda3/envs/satpy_himawari/lib/python3.8/site-packages/xarray/plot/plot.py in newplotfunc(darray, x, y, figsize, size, aspect, ax, row, col, col_wrap, xincrease, yincrease, add_colorbar, add_labels, vmin, vmax, cmap, center, robust, extend, levels, infer_intervals, colors, subplot_kws, cbar_ax, cbar_kwargs, xscale, yscale, xticks, yticks, xlim, ylim, norm, **kwargs)
    653             # Need the decorated plotting function
    654             allargs["plotfunc"] = globals()[plotfunc.__name__]
--> 655             return _easy_facetgrid(darray, kind="dataarray", **allargs)
    656 
    657         plt = import_matplotlib_pyplot()

~/tools/miniconda3/envs/satpy_himawari/lib/python3.8/site-packages/xarray/plot/facetgrid.py in _easy_facetgrid(data, plotfunc, kind, x, y, row, col, col_wrap, sharex, sharey, aspect, size, subplot_kws, ax, figsize, **kwargs)
    683 
    684     if kind == "dataarray":
--> 685         return g.map_dataarray(plotfunc, x, y, **kwargs)
    686 
    687     if kind == "dataset":

~/tools/miniconda3/envs/satpy_himawari/lib/python3.8/site-packages/xarray/plot/facetgrid.py in map_dataarray(self, func, x, y, **kwargs)
    284                 self._mappables.append(mappable)
    285 
--> 286         self._finalize_grid(x, y)
    287 
    288         if kwargs.get("add_colorbar", True):

~/tools/miniconda3/envs/satpy_himawari/lib/python3.8/site-packages/xarray/plot/facetgrid.py in _finalize_grid(self, *axlabels)
    388             self.set_axis_labels(*axlabels)
    389             self.set_titles()
--> 390             self.fig.tight_layout()
    391 
    392             for ax, namedict in zip(self.axes.flat, self.name_dicts.flat):

~/tools/miniconda3/envs/satpy_himawari/lib/python3.8/site-packages/matplotlib/cbook/deprecation.py in wrapper(*inner_args, **inner_kwargs)
    409                          else deprecation_addendum,
    410                 **kwargs)
--> 411         return func(*inner_args, **inner_kwargs)
    412 
    413     return wrapper

~/tools/miniconda3/envs/satpy_himawari/lib/python3.8/site-packages/matplotlib/figure.py in tight_layout(self, renderer, pad, h_pad, w_pad, rect)
   2611                else suppress())
   2612         with ctx:
-> 2613             kwargs = get_tight_layout_figure(
   2614                 self, self.axes, subplotspec_list, renderer,
   2615                 pad=pad, h_pad=h_pad, w_pad=w_pad, rect=rect)

~/tools/miniconda3/envs/satpy_himawari/lib/python3.8/site-packages/matplotlib/tight_layout.py in get_tight_layout_figure(fig, axes_list, subplotspec_list, renderer, pad, h_pad, w_pad, rect)
    301                               (colNum2 + 1) * div_col - 1))
    302 
--> 303     kwargs = auto_adjust_subplotpars(fig, renderer,
    304                                      nrows_ncols=(max_nrows, max_ncols),
    305                                      num1num2_list=num1num2_list,

~/tools/miniconda3/envs/satpy_himawari/lib/python3.8/site-packages/matplotlib/tight_layout.py in auto_adjust_subplotpars(fig, renderer, nrows_ncols, num1num2_list, subplot_list, ax_bbox_list, pad, h_pad, w_pad, rect)
     82             if ax.get_visible():
     83                 try:
---> 84                     bb += [ax.get_tightbbox(renderer, for_layout_only=True)]
     85                 except TypeError:
     86                     bb += [ax.get_tightbbox(renderer)]

~/tools/miniconda3/envs/satpy_himawari/lib/python3.8/site-packages/matplotlib/axes/_base.py in get_tightbbox(self, renderer, call_axes_locator, bbox_extra_artists, for_layout_only)
   4201                     # this artist
   4202                     continue
-> 4203             bbox = a.get_tightbbox(renderer)
   4204             if (bbox is not None
   4205                     and 0 < bbox.width < np.inf

~/tools/miniconda3/envs/satpy_himawari/lib/python3.8/site-packages/matplotlib/artist.py in get_tightbbox(self, renderer)
    284             if clip_path is not None and bbox is not None:
    285                 clip_path = clip_path.get_fully_transformed_path()
--> 286                 bbox = Bbox.intersection(bbox, clip_path.get_extents())
    287         return bbox
    288 

~/tools/miniconda3/envs/satpy_himawari/lib/python3.8/site-packages/matplotlib/path.py in get_extents(self, transform, **kwargs)
    601                 xys.append(curve([0, *dzeros, 1]))
    602             xys = np.concatenate(xys)
--> 603         return Bbox([xys.min(axis=0), xys.max(axis=0)])
    604 
    605     def intersects_path(self, other, filled=True):

~/tools/miniconda3/envs/satpy_himawari/lib/python3.8/site-packages/numpy/core/_methods.py in _amin(a, axis, out, keepdims, initial, where)
     41 def _amin(a, axis=None, out=None, keepdims=False,
     42           initial=_NoValue, where=True):
---> 43     return umr_minimum(a, axis, None, out, keepdims, initial, where)
     44 
     45 def _sum(a, axis=None, dtype=None, out=None, keepdims=False,

ValueError: zero-size array to reduction operation minimum which has no identity

```

The version information of some libraries as follows:

```
xarray version : 0.17.0
numpy version : 1.19.1
matplotlib version : 3.3.1
```

Any suggestions would be appreciated.

**Comment by andersy005** [comment]
@bugsuse, I think this is a bug in `matplotlib v3.3.1`, and I can confirm that everything appears to work with a more recent version of matplotlib ( I tested this on `matplotlib v3.3.4`). So, I'd recommend upgrading to any version after `v3.3.1`. I Hope this helps.

**Comment by bugsuse** [comment]
@andersy005 Yeah, you're right! Solved it by upgrading matplotlib to `v3.3.4`.  Thanks a lot!
